From a06f6a47938ec48934434f8a0ff379bf8f454944 Mon Sep 17 00:00:00 2001
From: David Leadbeater <dgl@dgl.cx>
Date: Wed, 3 Aug 2022 17:29:51 +1000
Subject: [PATCH] Add app-crypt/gpgme

Based on upstream Gentoo ebuild, with Python support removed, as that
isn't needed for cri-o.
---
 app-crypt/gpgme/Manifest               |   2 +
 app-crypt/gpgme/gpgme-1.17.1-r1.ebuild | 100 +++++++++++++++++++++++++
 app-crypt/gpgme/metadata.xml           |  11 +++
 3 files changed, 113 insertions(+)
 create mode 100644 app-crypt/gpgme/Manifest
 create mode 100644 app-crypt/gpgme/gpgme-1.17.1-r1.ebuild
 create mode 100644 app-crypt/gpgme/metadata.xml

diff --git a/app-crypt/gpgme/Manifest b/app-crypt/gpgme/Manifest
new file mode 100644
index 000000000..228275f1e
--- /dev/null
+++ b/app-crypt/gpgme/Manifest
@@ -0,0 +1,2 @@
+DIST gpgme-1.17.1.tar.bz2 1747692 BLAKE2B 44cf9309cbad038f23aa7c5c77fd85be23faf3d8e1eb7abdb87a737ad1ade1ee269a95ce618dba327bbd06e6f4b2a91e2446744ee08f09ea6fb3b2dad44431c5 SHA512 e6399c3de1e430e38f2692bf5ec0c02ecb36ea3dbb56ff29dc3a438a5be4900a77a0559dc5b673dc1ffbff5e7f589e548e19176b2644fe8f63e00c6b9181b920
+DIST gpgme-1.17.1.tar.bz2.sig 119 BLAKE2B 31832424c0e029dc520d3302a71b086d9fcb044c74344cc0fb4b6c6a02d0dd840e62e7eb60acfaf004e6f3650d58436b58e645ef2a50b948ecaf3ca162bf577a SHA512 e1a4213d0d39f2efd0d5c007a0449c76bd650e76e002a9acb673eef0579912db06d844df11a4f7cb821e1d8468d2d6c1303f1c5d915a5c3c9cb5abacfbd004fc
diff --git a/app-crypt/gpgme/gpgme-1.17.1-r1.ebuild b/app-crypt/gpgme/gpgme-1.17.1-r1.ebuild
new file mode 100644
index 000000000..a86cfdbcb
--- /dev/null
+++ b/app-crypt/gpgme/gpgme-1.17.1-r1.ebuild
@@ -0,0 +1,100 @@
+# Copyright 1999-2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+VERIFY_SIG_OPENPGP_KEY_PATH="${BROOT}"/usr/share/openpgp-keys/gnupg.asc
+
+inherit flag-o-matic qmake-utils toolchain-funcs verify-sig
+
+DESCRIPTION="GnuPG Made Easy is a library for making GnuPG easier to use"
+HOMEPAGE="http://www.gnupg.org/related_software/gpgme"
+SRC_URI="mirror://gnupg/gpgme/${P}.tar.bz2
+	verify-sig? ( mirror://gnupg/gpgme/${P}.tar.bz2.sig )"
+
+LICENSE="GPL-2 LGPL-2.1"
+# Please check ABI on each bump, even if SONAMEs didn't change: bug #833355
+# Use e.g. app-portage/iwdevtools integration with dev-libs/libabigail's abidiff.
+# Subslot: SONAME of each: <libgpgme.libgpgmepp.libqgpgme>
+SLOT="1/11.6.15"
+KEYWORDS="~alpha amd64 arm arm64 hppa ~ia64 ~loong ~m68k ~mips ppc ppc64 ~riscv ~s390 sparc x86 ~x64-cygwin ~amd64-linux ~x86-linux ~ppc-macos ~x64-macos ~sparc-solaris ~sparc64-solaris ~x64-solaris ~x86-solaris"
+IUSE="common-lisp static-libs +cxx qt5 test"
+RESTRICT="!test? ( test )"
+
+# Note: On each bump, update dep bounds on each version from configure.ac!
+RDEPEND=">=app-crypt/gnupg-2
+	>=dev-libs/libassuan-2.5.1:=
+	>=dev-libs/libgpg-error-1.36:=
+	qt5? ( dev-qt/qtcore:5 )"
+	#doc? ( app-doc/doxygen[dot] )
+DEPEND="${RDEPEND}
+	test? (
+		qt5? ( dev-qt/qttest:5 )
+	)"
+BDEPEND="verify-sig? ( sec-keys/openpgp-keys-gnupg )"
+
+REQUIRED_USE="qt5? ( cxx )"
+
+src_prepare() {
+	default
+
+	elibtoolize
+
+	# bug #697456
+	addpredict /run/user/$(id -u)/gnupg
+
+	local MAX_WORKDIR=66
+	if use test && [[ "${#WORKDIR}" -gt "${MAX_WORKDIR}" ]]; then
+		eerror "Unable to run tests as WORKDIR='${WORKDIR}' is longer than ${MAX_WORKDIR} which causes failure!"
+		die "Could not run tests as requested with too-long WORKDIR."
+	fi
+
+	# Make best effort to allow longer PORTAGE_TMPDIR
+	# as usock limitation fails build/tests
+	ln -s "${P}" "${WORKDIR}/b" || die
+	S="${WORKDIR}/b"
+}
+
+src_configure() {
+	local languages=()
+
+	use common-lisp && languages+=( "cl" )
+	use cxx && languages+=( "cpp" )
+	if use qt5; then
+		languages+=( "qt" )
+		#use doc ||
+		export DOXYGEN=true
+		export MOC="$(qt5_get_bindir)/moc"
+	fi
+
+	# bug #847955
+	append-lfs-flags
+
+	# bug #811933 for libassuan prefix
+	econf \
+		$(use test || echo "--disable-gpgconf-test --disable-gpg-test --disable-gpgsm-test --disable-g13-test") \
+		--enable-languages="${languages[*]}" \
+		--with-libassuan-prefix="${ESYSROOT}"/usr \
+		$(use_enable static-libs static)
+
+}
+
+src_compile() {
+	default
+}
+
+src_test() {
+	default
+}
+
+src_install() {
+	default
+
+
+	find "${ED}" -type f -name '*.la' -delete || die
+
+	# Backward compatibility for gentoo
+	# (in the past, we had slots)
+	dodir /usr/include/gpgme
+	dosym ../gpgme.h /usr/include/gpgme/gpgme.h
+}
diff --git a/app-crypt/gpgme/metadata.xml b/app-crypt/gpgme/metadata.xml
new file mode 100644
index 000000000..12645d415
--- /dev/null
+++ b/app-crypt/gpgme/metadata.xml
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE pkgmetadata SYSTEM "https://www.gentoo.org/dtd/metadata.dtd">
+<pkgmetadata>
+	<maintainer type="project">
+		<email>base-system@gentoo.org</email>
+		<name>Gentoo Base System</name>
+	</maintainer>
+	<use>
+		<flag name="common-lisp">Install common-lisp files</flag>
+	</use>
+</pkgmetadata>
-- 
2.37.1

